components/eam/src/chemistry/aerosol/mo_setsox.F90

```diff
@@ -777,81 +897,96 @@ subroutine SETSOX( &
                 patm_x = 1._r8
              endif
 
-             if (modal_aerosols) then
+!++hybrown
+          do jso4 = 1, nso4
+!-hybrown
 
-                pso4 = rah2o2 * 7.4e4_r8*EXP(6621._r8*work1(i)) * h2o2g * patm_x &
-                     * 1.23_r8 *EXP(3120._r8*work1(i)) * so2g * patm_x
+             if (modal_aerosols) then
```

Comment:
Just a general comment, I think in general it's better to try to keep these logicals out of loops. The performance difference would be negligible here since nso4 is small, but just something to keep in mind.